tests/support/headers.py

```diff
@@ -14,7 +15,38 @@
 )
 
 if TYPE_CHECKING:
-    from typing import Dict, List
+    from typing import Dict, List, Optional
+
+
+@dataclass
+class LinkData:
+    """Holds the data returned in an RFC 8288 ``Link`` header."""
+
+    prev_url: Optional[str]
+    """The URL of the previous page, or `None` for the first page."""
+
+    next_url: Optional[str]
+    """The URL of the next page, or `None` for the last page."""
+
+    first_url: str
+    """The URL of the first page."""
+
+
+def parse_link(header: str) -> LinkData:
```

Comment:
Could be a class method of `LinkData` (you can tell I'm grasping at things to suggest here).